manticore/wasm/structure.py

```diff
@@ -500,16 +587,68 @@ def grow(self, n: int) -> bool:
         :param n: The number of pages to attempt to add
         :return: True if the operation succeeded, otherwise False
         """
-        assert len(self.data) % 65536 == 0
-        ln = n + (len(self.data) // 65536)
+        ln = n + self.npages
         if ln > (2 ** 16):
             return False
         if self.max is not None:
             if ln > self.max:
                 return False
-        self.data.extend(0x0 for _ in range(n * 65536))  # TODO - these should also be symbolic
+        self._current_size = ln
         return True
 
+    def write_int(self, base, expression, size=32):
+        """
+        Writes an integer into memory.
+
+        :param base: Index to write at
+        :param expression: integer to write
+        :param size: Optional size of the integer
+        """
+        b = [
+            Operators.CHR(Operators.EXTRACT(expression, offset, 8)) for offset in range(0, size, 8)
+        ]
+        self.write_bytes(base, b)
+
+    def write_bytes(self, base, data):
+        """
+        Writes  a stream of bytes into memory
+
+        :param base: Index to start writing at
+        :param data: Data to write
+        """
+        self._publish("will_write_memory", base, base + len(data), data)
+        for idx, v in enumerate(data):
+            self._write_byte(base + idx, v)
+        self._publish("did_write_memory", base, data)
+
+    def read_int(self, base, size=32):
+        """
+        Reads bytes from memory and combines them into an int
+
+        :param base: Address to read the int from
+        :param size: Size of the int (in bits)
+        :return: The int in question
+        """
+        return Operators.CONCAT(
+            size, *map(Operators.ORD, reversed(self.read_bytes(base, size // 8)))
+        )
+
+    def read_bytes(self, base, size):
```

Comment:
Nit: Possible to add type annotations to these functions?

Reply:
Done. The type hints only describe concrete argument, which is fine since that's all we support right now. If we add a symbolic memory model we'll need to update them.